lib/Models/registerCustomComponentTypes.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+/*global require*/
+
+var customComponents = require('./customComponents');
+var CustomComponentType = require('./CustomComponentType');
+var CollapsibleViewModel = require('../ViewModels/CollapsibleViewModel');
+
+/**
+ * Registers custom component types.
+ * 
+ * @param  {Object} [options] Options.
+ * @param  {ChartPanelViewModel} [options.chartPanelViewModel] The chartPanelViewModel, if available.
+ */
+var registerCustomComponentTypes = function(options) {
+
+    var collapsibleComponentType = new CustomComponentType({
+        name: 'collapsible',
+        render: function(element) {
+            var collapsibleViewModel = new CollapsibleViewModel(element.attributes.name, element.attributes.open);
```

Comment:
This logic will be pretty much the same for all components, right?  Wouldn't something like this be sufficient?

```
customComponents.register('collapsible', CollapsibleViewModel);
```

The render would always just pass `element.attributes` to the constructor.